network/src/commonMain/kotlin/com/wire/kalium/network/KaliumKtorCustomLogging.kt

```diff
@@ -240,6 +249,49 @@ public class KaliumKtorCustomLogging private constructor(
     }
 }
 
+@Suppress("TooGenericExceptionCaught")
+private fun obfuscateAndLogMessage(text: String): String {
+    return try {
+        val obj = (Json.decodeFromString(text) as JsonElement)
+        if (obj.jsonArray.size > 0) {
+            obj.jsonArray.map {
+                obfuscateJsonElement(it)
+            }
+        } else {
+            obfuscateJsonElement(obj)
+        }
+        obj.toString()
+    } catch (e: Exception) {
+        "error the body content while logging "
+    }
+}
+
+fun obfuscateJsonElement(obj: JsonElement) {
```

Comment:
This doesn't `obfuscateJsonElement`, but rather `logObfuscatedJsonElement`

```suggestion
fun logObfuscatedJsonElement(obj: JsonElement) {
```